### PR TITLE
Fix NPE in KinesisSupervisor#setupRecordSupplier.

### PR DESCRIPTION
**Pull request by gianm**
PR #13539 refactored record supplier creation and introduced a bug: this method would throw NPE when recordsPerFetch was not provided by the user. recordsPerFetch isn't needed in this context at all, since the supervisor-side supplier doesn't fetch records. So this patch sets it to zero.

### CODE REVIEW

**Review by abhishekrb19 — APPROVED** [review]
LGTM, thanks!

**Comment by gianm** [comment]
> Not sure why checkstyle is failing with errors in code that's not touched in this PR:

That's an issue with master, will be fixed by #13860. I'll merge that back here when it's committed.

File: extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisor.java [hunk 1]
```diff
@@ -197,7 +197,7 @@ protected RecordSupplier<String, String, ByteEntity> setupRecordSupplier() throw
             ioConfig.getAwsAssumedRoleArn(),
             ioConfig.getAwsExternalId()
         ),
-        ioConfig.getRecordsPerFetch(),
+        0, // no records-per-fetch, it is not used
```
Comment (kfaraz) [inline extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisor.java:200]: Maybe rephrase the comment for clarity:
```suggestion
        0, // need not specify recordsPerFetch as supervisor-side supplier does not fetch records
```

**Review by kfaraz — APPROVED** [review]
Minor suggestion, otherwise LGTM

**Comment by gianm** [comment]
> Minor suggestion, otherwise LGTM

Thanks for the review. I merged it to avoid the need to re-run CI; would definitely 👍 a PR improving the comment, though.
